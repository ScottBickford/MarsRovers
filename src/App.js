import React from "react";
import "./App.css";
import { Container } from "semantic-ui-react";
import RoverInstructionsPage from "./features/roverInstructions/RoverInstructionsPage";

function App() {
  return (
    <Container className='main'>
      <h2>Rover Instructions</h2>
      <RoverInstructionsPage />
    </Container>
  );
}

export default App;
