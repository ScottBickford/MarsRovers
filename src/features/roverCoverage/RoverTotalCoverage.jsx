import React from 'react'
import { RoverCoverageTable } from './RoverCoverageTable';

export const RoverTotalCoverage = ({ roverMovements }) => {
  const positions = [];
  roverMovements.rovers.forEach(rover => {
    positions.push(...rover.positions);
  })
  return (
    <div style={{ paddingTop: 15 }}>
      <label>Total Coverage</label>
      <RoverCoverageTable upperRightCoords={roverMovements.upperRightCoords} positions={positions} />
    </div>
  )
}
