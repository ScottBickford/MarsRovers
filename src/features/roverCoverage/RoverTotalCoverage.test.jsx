import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { roverMovementsTestObject } from '../../App.test';
import { RoverTotalCoverage } from './RoverTotalCoverage';

describe('RoverInstructionsInput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoverTotalCoverage roverMovements={roverMovementsTestObject} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Input', () => {
    const { getByText } = render(<RoverTotalCoverage roverMovements={roverMovementsTestObject} />);
    const buttonElement = getByText(/Total Coverage/i);
    expect(buttonElement).toBeInTheDocument();
  });
});