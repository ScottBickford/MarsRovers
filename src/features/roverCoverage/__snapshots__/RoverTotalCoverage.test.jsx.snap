// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsInput snapshot renders 1`] = `
<div
  style={
    Object {
      "paddingTop": 15,
    }
  }
>
  <label>
    Total Coverage
  </label>
  <table
    className="ui celled collapsing table"
  >
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "lightblue",
            }
          }
        />
      </tr>
      <tr
        className=""
      >
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
        <td
          className=""
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;
